app/src/main/java/com/orange/ods/app/ui/utilities/composable/CodeImplementation.kt

```diff
@@ -262,6 +267,9 @@ class ListParameterValueBuilder {
 
     private val functions = mutableListOf<Function>()
 
+    inline fun <reified T> classInstance(noinline parameters: ParametersBuilder.() -> Unit = {}) =
+        classInstance(T::class.java, parameters)
+
     fun classInstance(clazz: Class<*>, parameters: ParametersBuilder.() -> Unit = {}) = apply { functions.add(ClassInstance(clazz, parameters)) }
```

Comment:
Do we have to keep this method which is only used by the other `classInstance` method?

Reply:
We can't directly call `apply { functions.add(ClassInstance(clazz, parameters)) }` in the inline method because `functions` is private. Thus we need to keep this method.

We could set it private but we can't because it's called in the inline method. In that case the proper way is to set the method `internal` and to add the `@PublishedApi` annotation to indicate that the method should be private but is internal because it is used in an inline method:

```kotlin
@PublishedApi
internal fun classInstance(clazz: Class<*>, parameters: ParametersBuilder.() -> Unit = {}) = apply { functions.add(ClassInstance(clazz, parameters)) }
```

But adding the `internal` keyword does not change anything because this method is defined in the `app` module.